service/src/main.rs

```diff
@@ -174,9 +176,23 @@ fn main() {
 		let skip_ra = smatches.is_present("skip-ra");
 		let dev = smatches.is_present("dev");
 
+		if clean_reset {
+			setup::initialize_shard_and_keys(enclave.as_ref(), &shard).unwrap();
+		}
+
 		let node_api =
 			node_api_factory.create_api().expect("Failed to create parentchain node API");
 
+		let request_state = smatches.is_present("request-state");
+		if request_state {
+			sync_state::sync_state(
+				&node_api,
+				&extract_shard(smatches, enclave.as_ref()),
+				enclave.as_ref(),
+				skip_ra,
+			);
+		}
```

Comment:
Request-state can happen here, before starting the worker - does not need to be a separate call anymore.